server/src/internalClusterTest/java/org/opensearch/indices/stats/IndexStatsIT.java

```diff
@@ -288,6 +288,33 @@ public void testFieldDataStats() throws InterruptedException {
 
     }
 
+    public void testLastIndexRequestTimestamp() throws Exception {
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareCreate("test")
+                .setSettings(settingsBuilder().put("index.number_of_replicas", 0).put("index.number_of_shards", 2))
+                .setMapping("field", "type=text,fielddata=true")
+                .get()
+        );
+        assertAcked(client().admin().indices().prepareCreate("test").setSettings(settingsBuilder()).get());
+        ensureGreen();
+
+        // Index a document
+        client().prepareIndex("test").setId("1").setSource("field", "value1").setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE).get();
+        IndicesStatsResponse statsResponse = client().admin().indices().prepareStats("test").get();
+        IndexStats stats = statsResponse.getIndex("test");
+        long ts1 = stats.getLastIndexRequestTimestamp();
+        assertTrue("Timestamp should be set after first write", ts1 > 0);
+
+        // Wait and index another document
+        Thread.sleep(1000);
```

Comment:
Fixed sleeps are almost always bad. In this case it makes the test slower than it needs to be, and technically there's no guarantee that the underlying system clock will actually change within 1 second (though that's unlikely to be a problem in practice here). You can use the `assertBusy(() -> ...)` helper to execute the following code in a loop with back off and retry. It will almost always pass on the first iteration (much faster than 1 second), but in case the test machine clock is behaving weird it will still work as expected.